### ISSUE TITLE
make var mixer accessible without need of channel assigment

### ISSUE THREAD

**Issue by strgaltdel** [FrSkyRC/ETHOS-Feedback-Community#2270]
at first,
a happy new year,
health and luck,
and always bugfree code !

Var mixers are a very good way to configure mixer parameters, especially if the variable is spread over several mixers.
Another advantage is that you can centralize the model configuration / adjustment without having to call up the individual mixers, scroll to the parameters and change them.
all in all very efficient!
(among some other points I didn't mention).

On the other hand, var mixers (when used as a simple variable) are doomed to consume an entire channel so they can be accessed by mixers.
I know that we have twice as many mixers as in Otx, but in a complex environment, and if you want to group mixers, have some headroom between those mixers, etc... 
you can get into the situation where you want to have some more Var mixers but you don't have any real free channels.

My suggestion would be to introduce a similar handling as with Flightmodes to get an almost "unlimited" access to simple Var mixers:
Flightmodes are not accessible in categories if one has not declared any.
If one has no flight modes (default only), there is no entity.
But as soon you add a FM, you'll have a new category in your choice list... 

So:
If a var mixer is not assigned to a channel 
(i.e. only used as a value without further calculations in a channel)
>> make it accessible in its own category like "simple value" for the mixer parameter wherever you can choose a source.

This would have the following advantages
- Use of var's without the need to allocate channels (more free channels)
- theoretically unlimited number of var's
- Ability for users to configure most model-specific mixer parameters in one menu (Var mixer menu) without having to jump between dozens of mixers and scroll to the relevant parameter.
- and as a future step, if these var mixers can be accessed via lua, it allows lua devs to develop GUIs with guidance for the user, so that one could configure even more complex models through simplified lua widgets.
....

hope it won't be too much hard work....

thanks and regards
udo

**Comment by hrenz** [comment]
this is then nearly like a global Variabel

**Comment by azaz44** [comment]
I was configuring some models recently and I agree with the OP. Having to map mixers to channels to access them as "source" is completely not in spirit of Ethos. This way we're coming back to OpenTX, where you have so many channels, so many LS, so many curves etc. and they exist no matter if you use them or not. No way to delete, reorder, or get any idea what is used and what is not.

You maybe want to have a "VAR" for rudder expo. You have to bind it to some channels, you have to select it - so it is maybe 40. Then in another model you see, that channel 40 is busy, so you need to pick up some other number. And all the mess begins.

Not to mention that channels have 2 values (before and after recenting / mapping) and it's confusing what is used.

In addition, I think it would be really beneficial to have a separate screen to configure your values. So mixers do the logic, but what are values of rate, expo, whatever in each mode would be on some other screen. So you could tweak these values without going to heavy logic of the mixers. So we kind of come back to global variables, but in form of var mixers, which are very nice...

**Comment by strgaltdel** [comment]
it seems that we have about the same idea in the long run.

**- The first step would be to "separate" vars and channels (as an option), which would allow a better structure and a higher number of vars.**

a higher number of variables allows to make adjustments by simply selecting the right variable without having to browse / navigate through (more or less complex) mixer structures



**- the second step would be to introduce a Lua method to get access to the variables via scripts**

This would allow the development of widgets for adjusting a whole set of parameters on the fly by selecting a particular variable, fine tuning it, saving it, and selecting the next variable using only a small number of inputs (one switch for selecting the variable and saving the value, one pod for adjusting it). I did this for oTx and it worked much better than in-flight trim, using dedicated inputs for each parameter.



**- The third step would be to develop a Lua user interface so that a "normal" user can easily go through the settings of a complex model.**

Say we have a UI that configures 20..30 well defined parameters :
experienced users in the community could design model templates using these variables and publish them
The less experienced users would not have to work through a more complex template design to achieve their personal setup, they would just have to make adjustments at a higher level on a (best case standardized) UI



... just my thoughts when I opened the topic
